### PR TITLE
Fix search service for non-page indexes

### PR DESCRIPTION
**Pull request by rpribyl**

### CODE REVIEW

File: src/DancingGoat/Controllers/SearchController.cs [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
+        }
+
+
+        // GET: Search
+        [ValidateInput(false)]
+        public ActionResult Index(string searchText, int? page)
+        {
+            int totalItemsCount;
+            var pageNumber = (page ?? 1);
+            pageNumber = Math.Max(pageNumber, 1);
```
Comment (ondrejsevcik) [inline src/DancingGoat/Controllers/SearchController.cs]: 3x recalculating pageNumber. It's a bit confusing. Could it be simplified?
If the pageNumber = 0 means the first page, we should add some explanation comment.

File: src/Kentico.Search/SearchResultItem.cs [hunk 2]
```diff
@@ -38,6 +39,16 @@ public string Content
 
 
         /// <summary>
+        /// Gets the image GUID associated with the search item. Contains data from the field configured as 'Image field' in search index configuration.
+        /// </summary>
+        public Guid Image
```
Comment (ondrejsevcik) [inline src/Kentico.Search/SearchResultItem.cs]: I would expect that property Image will serve Image or something of type Image. IMHO it should be named more clearly like ImageGuid.

File: src/Kentico.Search/SearchService.cs [hunk 3]
```diff
@@ -53,13 +54,13 @@ public SearchService(string[] searchIndexNames, string cultureName, string siteN
...
-        /// <param name="page">Zero-based page index</param>
-        /// <param name="pageSize">Specifies the number of results per page</param>
-        /// <param name="numberOfResults">Total number of search results</param>
-        /// <returns>Search results or null if <paramref name="query"/> is empty.</returns>
+        /// <param name="query">Text to search.</param>
+        /// <param name="page">Zero-based page index.</param>
+        /// <param name="pageSize">Specifies the number of results per page. Page size must be a positive number.</param>
+        /// <param name="numberOfResults">Total number of search results for the given query which can be retrieved from the search index.</param>
+        /// <returns>Search results for the specified page or null if <paramref name="query"/> is empty.</returns>
         public virtual IEnumerable<SearchResultItem> Search(string query, int page, int pageSize, out int numberOfResults)
```
Comment (ondrejsevcik) [inline src/Kentico.Search/SearchService.cs:64]: Parameter page should be named pageIndex to match SearchInternal method. It's also clearer what is this parameter about. 
(If it's not a breaking change).

File: test/Kentico.Search.Tests/Fakes/FakeSearchService.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+using CMS.Base;
+using CMS.DocumentEngine;
+using CMS.Search;
+
+namespace Kentico.Search.Tests
+{
+    public class FakeSearchService : SearchService
+    {
+        private string _dataSetFilePath = null;
```
Comment (ondrejsevcik) [inline test/Kentico.Search.Tests/Fakes/FakeSearchService.cs]: The other classes use `m` prefix for private variables. We should be consistent.

**Comment by ondrejsevcik** [comment]
Just a side note:
Understanding tests was a bit harder as it contains a lot of fake objects so I wasn't sure what's actually tested. We should add some test coverage support to see what's actually tested.

File: src/DancingGoat/Controllers/SearchController.cs (hunk 1 above)
Comment (rpribyl) [inline src/DancingGoat/Controllers/SearchController.cs]: Yep, I've simplified it a bit. I guess, comment is no longer needed

File: src/Kentico.Search/SearchResultItem.cs (hunk 2 above)
Comment (rpribyl) [inline src/Kentico.Search/SearchResultItem.cs]: Makes sense. ImageGuid fits its purpose much better.

File: src/Kentico.Search/SearchService.cs (hunk 3 above)
Comment (rpribyl) [inline src/Kentico.Search/SearchService.cs:64]: I've kept the "page" param exactly due to backward compatibility. If we broke it, we would have to increase the package major version (according to the semantic versioning: http://semver.org/)
I plan to refactor the search service a bit in the next major version and fix the page number parameter to on-based anyway.

File: test/Kentico.Search.Tests/Fakes/FakeSearchService.cs (hunk 4 above)
Comment (rpribyl) [inline test/Kentico.Search.Tests/Fakes/FakeSearchService.cs]: Fixed

**Pull request merged**